### DISCUSSION TITLE
Use Mockito to mock DAOs?

### DISCUSSION THREAD

**Discussion by madsane29** [simolus3/drift#2105] [Q&A]
Is it possible? May I have an example to make it work?

I need some unit tests, I have a lot of database calls and I'd need to mock the database/dao but no success.

I wonder if there is some hidden example that is not in the docs?

**Comment by simolus3** [comment]
It is much, much easier to just manually insert some entries and then run your unit test on that. You can use a drift database in unit tests if sqlite3 libraries are installed on your computer. So I recommend using a real database bound to `NativeDatabase.memory()` in unit tests. Then you have a real database and DAO with no need to mock anything.

I don't have an example for a mocked drift database or DAO. Drift hasn't been designed to be easy to mock. Internally in drift, I'm mocking the  `QueryExecutor` (which can be used as a replacement for `NativeDatabase`): https://github.com/simolus3/drift/blob/3253cd7ead51d7b163542929eb0fc8daee573fce/drift/test/test_utils/mocks.dart#L5-L119

But really, I strongly recommend just using a real, in-memory database.

**Reply by kuhnroyal** [reply]
It is pretty easy to do with Mockito.

```
customMocks: [
  MockSpec<MyDao>(unsupportedMembers: {#alias}),
]
```

Same for the database class.
